pkg/tinykv/tinykv.go

```diff
@@ -129,6 +136,41 @@ func (kv *store[T]) Get(k string) (T, bool) {
 	return e.value, ok
 }
 
+func (kv *store[T]) GetWithTimeout(k string) (EntityWithTimeout, bool) {
+	kv.mx.Lock()
+	defer kv.mx.Unlock()
+
+	e, ok := kv.kv[k]
+	if !ok {
+		return EntityWithTimeout{}, ok
+	}
+	if e.expired() {
```

Comment:
I don't think this method should do any check on the timeouts and should not delete from the KV.

I think the method should be `GetExpiration` for a key.

Reply:
I don't fully agree with this. If I wanted to get a value and an expiration, under your suggestion I would need to call 2 separate functions, which COULD cause an issue whereby, I manage to retrieve the value, but in the time between calling `Get` and `GetExpiration` the key expired, it would cause an error because the Key would have expired. 

Keeping it as 1 call to get the value and expiration as a single struct makes more sense in my opinion. Why do you think we shouldn't be checking on timeout or deleting from KV like `Get` does?